specs/vep-2420-getting-started-with-my-data/README.md

```diff
@@ -0,0 +1,255 @@
+4. There is no IDE assistance. Frameworks like spring support autocomplete in their property and yaml files.
+5. Over reliance on using config files instead of users setting the values through python objects. Which would provide better error handling, documentation etc... It is more common practise when using a library/framework to  use less config file and most object initialization. The exception being spring.
+
+
+### Poor documentation and implementation around secrets and properties
+
+When first starting with VDK and your own data you will need to provide VDK with passwords/username/etc. to connect to you data source.
+
+#### Pain points
+The documentation completely omits how to handle this in the getting started section.
+After you have completed a getting started section it leaves a user still unsure how to write a production quality job.
+
+#### Root Causes
+1. There is a file based implementation of the properties' provider. This is not heavily enough advertised in the docs.
+2. Furthermore there is no file based implementation of the secrets provider. So on even the most trivial example of reading data the user is left wondering how to handle secrets.
+
+### Not trivial to run in the IDE
+
+Alot of developers just want to copy a getting started into their IDE and click run.
+That will be used as a launch point to explore the product.
+
+#### Pain points
+Its not obvious how to run it in the IDE
+
+
+#### Root Causes
+1. There is actually great documentation on how to get this working. But it needs to be included as the first line of the getting started section as this is the first thing that should be done after creating a job.
+2. They don't call our product from a main. If they called it from a main we wouldn't need to explain how to run it.
+
+### Reliant on env variables
+
+In our tutorials we encourage users to set ingest destination using env variables: https://github.com/vmware/versatile-data-kit/wiki/Ingesting-data-from-DB-into-Database.
+
+#### Pain points
+1. How env variables are set is different on linux and windows. This results in our tutorial having less support for windows users as we write our tutorials assuming mac/linux users
+2. Works on my computer problems. If they complete a tutorial and check it into git, someone else won't be able to run it without knowing they need to set a list of env variables
+3. Poor IDE support. If they set the properties in the terminal and run examples they will see everything running ok. then they run it in their IDE and it won't work because it isn't picking up the properties. This is painful experince for developers
+4. They might restart their computer and the example won't run and they can't remember why
+
+
+## Requirements and goals
+
+1. Easily finding out which properties need to be set for a given task
```

Comment:
@antoniivanov  Let me know what you think.

The root point is a requirements and sub bullet points are SMART goals to measure it against.